### PR TITLE
Listen and process incoming `sig` messages

### PR DESCRIPTION
**Pull request by darosior**
This builds on top of #2 to implement the server for the client (the stakeholder) to have us watch for specific vaults. The listener loop is kept simple. A functional test suite is added to start testing paths not really accessible through unit tests.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1,5 +1,668 @@
...
+source = "git+https://github.com/revault/revault_tx?branch=master#695415edc410e4420884b3a7a497139b7fece7c1"
+dependencies = [
+ "base64",
+ "bitcoinconsensus",
+ "miniscript",
+ "serde",
+]
+
+[[package]]
+name = "revault_tx"
```
Comment (JSwambo) [inline Cargo.lock:421]: `revault_tx` specified twice in the lock file so it doesn't compile for me

File: src/listener.rs [hunk 2]
```diff
@@ -0,0 +1,947 @@
...
+        match self {
+            Self::Io(ref e) => write!(f, "io error: '{}'", e),
+            Self::Db(ref e) => write!(f, "database error: '{}'", e),
+            Self::Tx(ref e) => write!(f, "transaction handling error: '{}'", e),
+            Self::BitcoinD(ref e) => write!(f, "bitcoind communication error: '{}'", e),
+            Self::UnknownOutpoint(ref o) => write!(f, "unknown outpoint: '{}'", o),
+            Self::UnknownTxid(ref t) => write!(f, "unknown txid: '{}'", t),
+            Self::UnexpectedEmerSig(ref o) => write!(f, "received an Emergency signature for an \
+                                                         existing vault ('{}')", o),
+            Self::UnexpectedUnEmerSig(ref o) => write!(f, "received an UnvaultEmergency signature for a \
```
Comment (JSwambo) [inline src/listener.rs:48]: todo: when running with emergency feature... expect emergency sigs first

**Comment by JSwambo** [comment]
Ack  up to 227f9a380ba8138a7a9cd6511e0087c280679980 for now.

Gonna take some time to go through the test framework, will try to finish review tomorrow.

**Comment by JSwambo** [comment]
How about adding a little README for the txbuilder tool?

**Comment by darosior** [comment]
As noted in the commit, it is a hack not meant to be used by anyone so i'd rather not add README

**Comment by darosior** [comment]
Ok, sorry for the force push noise. Was checking CI runs.

Added a few CI jobs, and rebased on master.

File: tests/test_framework/bitcoind.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        wait_for(lambda: self.rpc.getblockcount() == old_blockcount + numblocks)
+
+    def create_utxo(self, address, amount):
+        """Send {amount} coins to this {address}, return the outpoint of the
+        created UTXO."""
+        txid = self.rpc.sendtoaddress(address, amount)
+        vout = 0
+        while True:
+            txoinfo = self.rpc.gettxout(txid, vout)
+            if txoinfo["scriptPubKey"]["address"] == address:
```
Comment (JSwambo) [inline tests/test_framework/bitcoind.py:116]: `"address"` gave a `KeyError`. I changed to 
```python
if txoinfo["scriptPubKey"]["addresses"][0] == address:
```
to fix it (as per the [docs](https://bitcoincore.org/en/doc/0.21.0/rpc/blockchain/gettxout/))

**Comment by darosior** [comment]
You need to update to `22.0` https://bitcoincore.org/en/releases/22.0/
‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by darosior** [comment]
Not introduced in this PR but re-reading the code this is wrong:
```rust
pub fn db_revoc_confirmed(
    db_path: &path::Path,
    vault_id: i64,
    height: i32,
) -> Result<(), DatabaseError> {
    db_exec(&db_path, |db_tx| {
        db_tx.execute(
            "UPDATE vaults SET revoc_height = (?1) WHERE id = (?2)",
            params![height, vault_id],
        )?;

        Ok(())
    })
}
```

It should set `should_cancel` to `0`. Or maybe cleaner: have `db_canceling_vaults` rule out rows with a `revoc_height` set.

**Review by JSwambo — APPROVED** [review]
Ack a6b44ced2b3b65435944215d2e600fcd73c93fab

**Pull request merged**